packages/core/src/stateUtils.ts

```diff
@@ -806,17 +806,10 @@ export function transitionParallelNode<
       _event
     );
     if (nextStateNode) {
-      transitionMap[subStateKey] = nextStateNode;
+      transitions.push(...nextStateNode);
```

Comment:
Sigh, I didn't give this one the best name

```suggestion
      // TODO: rename nextStateNode to nextTransitions
      transitions.push(...nextStateNode);
```